src/featureFlags.ts

```diff
@@ -84,6 +85,27 @@ const getSubmitArticleState = (flags: IFlags, user: User) => {
   return flags?.submit_article;
 };
 
+interface UpdateParam {
+  feature: string;
+  replacement: Flag['value'];
+  onCheckValidity(value: Flag['value']): boolean;
+}
+
+export const adjustAnonymousFlags = (
+  flags: IFlags,
+  updates: UpdateParam[],
+): IFlags => {
+  const updatedFlags = { ...flags };
+
+  updates.forEach(({ feature, onCheckValidity, replacement }) => {
+    if (flags?.[feature]?.enabled && onCheckValidity(flags[feature].value)) {
+      updatedFlags[feature] = { enabled: true, value: replacement };
```

Comment:
If the user is not applicable for the flag, I suggest setting it to `enabled: false` at this point without value. This way we know in the data that this user is not part of an experiment